spring-integration-ip/src/main/java/org/springframework/integration/ip/dsl/TcpInboundChannelAdapterSpec.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.ip.dsl;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import org.springframework.integration.dsl.ComponentsRegistration;
+import org.springframework.integration.dsl.MessageProducerSpec;
+import org.springframework.integration.ip.tcp.TcpReceivingChannelAdapter;
+import org.springframework.integration.ip.tcp.connection.AbstractConnectionFactory;
+import org.springframework.scheduling.TaskScheduler;
+
+/**
+ * A {@link MessageProducerSpec} for {@link TcpReceivingChannelAdapter}s.
+ *
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public class TcpInboundChannelAdapterSpec
+		extends MessageProducerSpec<TcpInboundChannelAdapterSpec, TcpReceivingChannelAdapter>
+		implements ComponentsRegistration {
+
+	private final AbstractConnectionFactory connectionFactory;
+
+	TcpInboundChannelAdapterSpec(AbstractConnectionFactory connectionFactory) {
+		super(new TcpReceivingChannelAdapter());
+		this.connectionFactory = connectionFactory;
```

Comment:
Well, I see your point about `ComponentsRegistration`. 
How about to have alternative ctor for the `AbstractConnectionFactorySpec` and populate for registration just only result of that?
Any `AbstractConnectionFactory` variant should be already supplied as a `@Bean`
WDYT?

I suggests this because I don't know how to answer to your concern in the https://github.com/spring-projects/spring-integration-java-dsl/issues/137 :smile: